pkg/cmd/target/set.go

```diff
@@ -169,3 +189,99 @@ var TargetSetCmd = &cobra.Command{
 		return nil
 	},
 }
+
+func HandleTargetJSON(data []byte) error {
+	ctx := context.Background()
+
+	apiClient, err := apiclient_util.GetApiClient(nil)
+	if err != nil {
+		return err
+	}
+	serverConfig, res, err := apiClient.ServerAPI.GetConfigExecute(apiclient.ApiGetConfigRequest{})
+	if err != nil {
+		return apiclient_util.HandleErrorResponse(res, err)
+	}
+
+	providersManifest, err := manager.NewProviderManager(manager.ProviderManagerConfig{
+		RegistryUrl: serverConfig.RegistryUrl,
+	}).GetProvidersManifest()
+	if err != nil {
+		log.Error(err)
+	}
+
+	var latestProviders []apiclient.Provider
+	if providersManifest != nil {
+		providersManifestLatest := providersManifest.GetLatestVersions()
+		if providersManifestLatest == nil {
+			return errors.New("could not get latest provider versions")
+		}
+
+		latestProviders = provider.GetProviderListFromManifest(providersManifestLatest)
+	} else {
+		fmt.Println("Could not get provider manifest. Can't check for new providers to install")
+	}
+	providerList, err := provider.GetProviderViewOptions(apiClient, latestProviders, ctx)
+	if err != nil {
+		return err
+	}
+
+	selectedProvider, err := provider_view.GetProviderFromPrompt(providerList, "Choose a Provider", false)
+	if err != nil {
+		if common.IsCtrlCAbort(err) {
+			return nil
+		} else {
+			return err
+		}
+	}
+	if selectedProvider == nil {
+		return errors.New("no provider selected")
+	}
+	if selectedProvider.Installed != nil && !*selectedProvider.Installed {
+		if providersManifest == nil {
+			return errors.New("could not get providers manifest")
+		}
+		err = provider.InstallProvider(apiClient, *selectedProvider, providersManifest)
+		if err != nil {
+			return err
+		}
+	}
+
+	var selectedTarget *target_view.TargetView
+	err = target.ParseJSON(data, &selectedTarget)
+	if err != nil {
+		return fmt.Errorf("failed to parse input: %w", err)
+	}
+
+	if selectedTarget.Name == "" {
+		return errors.New("invalid input: 'name' field is required")
+	}
+	if selectedTarget.Options == "" {
+		return errors.New("option fields are required to setup your target")
+	}
+	targetManifest, res, err := apiClient.ProviderAPI.GetTargetManifest(ctx, selectedProvider.Name).Execute()
```

Comment:
everything above this point, in this function, can be removed.

There is no selecting providers. The target options have the provider info already stored. You need to use that.

Reply:
@Tpuljak please have a look at the changes i made . 
when i tested it out, i got this error 
`FATA[0000] provider not found: provider not found`

Reply:
Try to wait a bit first and make sure the providers are listed when running `dtn provider list`. It can take a few seconds for them to load

Reply:
please have look at this , i having some version conflicts, please could you try it out and see if it works properly
![target1](https://github.com/user-attachments/assets/c542483f-2b9d-403d-b8a0-70a0bc8819a9)

Reply:
> it seems that you have another instance of the Daytona server running somewhere.

Yh, its been resolved